controllers/pokemon_controller.go

```
@@ -0,0 +1,37 @@
+package controllers
+
+import (
+	"golangBootcamp/m/services"
+	"net/http"
+
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+func FindPokemons(c *gin.Context) {
+	pokemons, error := services.FindAllPokemons()
+	if error != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"message": error})
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"data": pokemons})
+}
+
+func FindPokemonById(c *gin.Context) {
+	id, error := strconv.Atoi(c.Param("id"))
+	if error != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": error})
```

When returning an error response to the user of your API you want to return something that isn't too explicit about how your API works. In other words, you don't want to return the exact error you got because that would be giving away too much info to the user.

What is usually done, is when you are handling an error, you only want to log the error you got, and then return something more custom to the user, like

```
log.Error(err)
c.JSON(http.StatusBadRequest, gin.H{"message": "something went wrong!"})
```

---

Of course, when you can be more descriptive you should, in this case that the ID was not an int, you could tell the user `invalid id param` or something like that

---

Done :D